### PR TITLE
feat: flexbile font icon usage in tab navigation

### PR DESCRIPTION
**Pull request by MartoYankov**
Enables more flexible declaration and manipulation of font icons as tab bar icons. Utilizes the nested Image and Label elements better.

Part of the 6.1 release for the Tab Navigation story - https://github.com/NativeScript/NativeScript/issues/6967

Fixes https://github.com/NativeScript/NativeScript/issues/7623
Fixes https://github.com/NativeScript/NativeScript/issues/7506
Fixes https://github.com/NativeScript/NativeScript/issues/7470
Fixes https://github.com/NativeScript/NativeScript/issues/7547

### CODE REVIEW

File: tns-core-modules-widgets/android/widgets/src/main/java/org/nativescript/widgets/BottomNavigationBar.java [hunk 1]
```diff
@@ -205,6 +205,19 @@ private void setupItem(LinearLayout ll, TextView textView,ImageView imgView, Tab
             textView.setVisibility(GONE);
         }
 
+        if (tabItem.typeFace != null) {
+            textView.setTypeface(tabItem.typeFace);
```
Comment (manoldonev) [inline tns-core-modules-widgets/android/widgets/src/main/java/org/nativescript/widgets/BottomNavigationBar.java]: Maybe set typeface, text size, and color only if textview is visible? (i.e. inside the "if" block above)

File: tns-core-modules-widgets/android/widgets/src/main/java/org/nativescript/widgets/TabsBar.java [hunk 2]
```diff
@@ -288,6 +287,19 @@ private void setupItem(LinearLayout ll, TextView textView,ImageView imgView, Tab
             ll.setBackgroundColor(tabItem.backgroundColor);
         }
 
+        if (tabItem.typeFace != null) {
```
Comment (manoldonev) [inline tns-core-modules-widgets/android/widgets/src/main/java/org/nativescript/widgets/TabsBar.java]: Same comment as for bottomnav above.

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 3]
```diff
@@ -167,50 +167,84 @@ function initializeNativeClasses() {
...
 
 function createTabItemSpec(tabStripItem: TabStripItem): org.nativescript.widgets.TabItemSpec {
-    let iconSource;
     const tabItemSpec = new org.nativescript.widgets.TabItemSpec();
 
-    // Image and Label children of TabStripItem
-    // take priority over its `iconSource` and `title` properties
-    iconSource = tabStripItem.image ? tabStripItem.image.src : tabStripItem.iconSource;
-    tabItemSpec.title = tabStripItem.label ? tabStripItem.label.text : tabStripItem.title;
+    if (tabStripItem.isLoaded) {
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: When is tabStripItem not loaded -- what happens then?

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 4]
```diff
@@ -167,50 +167,84 @@ function initializeNativeClasses() {
...
+        const nestedLabel = tabStripItem.label;
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: We should discuss this nested naming -- I think I know what you mean but at first glance I have a different association.

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 5]
```diff
@@ -167,50 +167,84 @@ function initializeNativeClasses() {
...
         }
     }
 
     return tabItemSpec;
 }
 
+function _getIcon(tabStripItem: TabStripItem): android.graphics.drawable.BitmapDrawable {
+    const iconSource = tabStripItem.image && tabStripItem.image.src;
+
+    let is = new ImageSource();
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: No need to initialize it here.

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 6]
```diff
@@ -167,50 +167,84 @@ function initializeNativeClasses() {
...
+                    // traceMissingIcon(iconSource);
+                }
             }
         }
     }
 
     return tabItemSpec;
 }
 
+function _getIcon(tabStripItem: TabStripItem): android.graphics.drawable.BitmapDrawable {
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: Leading underscore in the function name is unnecessary here (same as setElevation below)

File: tns-core-modules/ui/tab-navigation-base/tab-strip-item/tab-strip-item.ts [hunk 7]
```diff
@@ -9,7 +10,7 @@ import { AddChildFromBuilder } from "../../core/view";
 
 // Requires
 import { 
-    View, CSSType, backgroundColorProperty, backgroundInternalProperty, colorProperty, 
+    View, ViewBase, CSSType, backgroundColorProperty, backgroundInternalProperty, colorProperty,
```
Comment (manoldonev) [inline tns-core-modules/ui/tab-navigation-base/tab-strip-item/tab-strip-item.ts]: There are some unused imports here (we can clean them up at a later stage).

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts (hunk 6 above)
Comment (MartoYankov) [reply tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: Will refactor separately - atm it's named like this in the old implementations too.

**Pull request merged**